packages/flutter/lib/src/painting/image_provider.dart

```diff
@@ -683,28 +683,29 @@ abstract class AssetBundleImageProvider extends ImageProvider<AssetBundleImageKe
   }
 }
 
+/// Key for the image obtained by a [ResizeImage].
+///
+/// This is used to identify the precise resource in the [imageCache].
 @immutable
-class _SizeAwareCacheKey {
-  const _SizeAwareCacheKey(this.providerCacheKey, this.width, this.height);
-
-  final Object providerCacheKey;
+class ResizeImageKey {
+  const ResizeImageKey._(this._providerCacheKey, this._width, this._height);
```

Comment:
I guess making the constructor and fields private avoids the issue in practice...